controllers/tls/mongodb_tls.go

```diff
@@ -15,50 +16,66 @@ import (
 	"github.com/mongodb/mongodb-kubernetes-operator/pkg/kube/podtemplatespec"
 	"github.com/mongodb/mongodb-kubernetes-operator/pkg/kube/statefulset"
 
+	kubernetesClient "github.com/mongodb/mongodb-kubernetes-operator/pkg/kube/client"
+	"go.uber.org/zap"
 	apiErrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
 
-	mdbv1 "github.com/mongodb/mongodb-kubernetes-operator/api/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 const (
-	tlsCAMountPath             = "/var/lib/tls/ca/"
-	tlsCACertName              = "ca.crt"
-	tlsOperatorSecretMountPath = "/var/lib/tls/server/" //nolint
-	tlsSecretCertName          = "tls.crt"              //nolint
-	tlsSecretKeyName           = "tls.key"
-	tlsSecretPemName           = "tls.pem"
+	TLSCAMountPath             = "/var/lib/tls/ca/"
+	TLSCACertName              = "ca.crt"
+	TLSOperatorSecretMountPath = "/var/lib/tls/server/" //nolint
 )
 
+type TLSResource interface {
+	IsTLSEnabled() bool
+	IsTLSOptional() bool
+	TLSConfigMapNamespacedName() types.NamespacedName
+	TLSSecretNamespacedName() types.NamespacedName
+	NamespacedName() types.NamespacedName
+	TLSOperatorSecretNamespacedName() types.NamespacedName
+	GetOwnerReferences() []metav1.OwnerReference
+	TLSSecretKeyName() string
+	TLSSecretCertName() string
+	TLSSecretPEMName() string
+	TLSConfigMapRequiredEntries() []string
+}
+
 // validateTLSConfig will check that the configured ConfigMap and Secret exist and that they have the correct fields.
-func (r *ReplicaSetReconciler) validateTLSConfig(mdb mdbv1.MongoDBCommunity) (bool, error) {
-	if !mdb.Spec.Security.TLS.Enabled {
+func ValidateTLSConfig(mdb TLSResource, client kubernetesClient.Client, log *zap.SugaredLogger, secretWatcher *watch.ResourceWatcher) (bool, error) {
+	if !mdb.IsTLSEnabled() {
 		return true, nil
 	}
 
-	r.log.Info("Ensuring TLS is correctly configured")
+	log.Info("Ensuring TLS is correctly configured")
 
 	// Ensure CA ConfigMap exists
-	caData, err := configmap.ReadData(r.client, mdb.TLSConfigMapNamespacedName())
+	caData, err := configmap.ReadData(client, mdb.TLSConfigMapNamespacedName())
 	if err != nil {
 		if apiErrors.IsNotFound(err) {
-			r.log.Warnf(`CA ConfigMap "%s" not found`, mdb.TLSConfigMapNamespacedName())
+			log.Warnf(`CA ConfigMap "%s" not found`, mdb.TLSConfigMapNamespacedName())
 			return false, nil
 		}
 
 		return false, err
 	}
 
-	// Ensure ConfigMap has a "ca.crt" field
-	if cert, ok := caData[tlsCACertName]; !ok || cert == "" {
-		r.log.Warnf(`ConfigMap "%s" should have a CA certificate in field "%s"`, mdb.TLSConfigMapNamespacedName(), tlsCACertName)
-		return false, nil
+	// Ensure ConfigMap has all the required fields
+	for _, key := range mdb.TLSConfigMapRequiredEntries() {
+		if cert, ok := caData[key]; !ok || cert == "" {
```

Comment:
Enterprise requires two entries in the CM, so we need to add a loop which here will execute only once